spec/rest_client_failure_response_spec.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'rest-client'
+require 'rest_client_failure_response'
+
+RSpec.describe RestClientFailureResponse do
+  describe '#ok?' do
+    it 'returns false' do
+      custom_response = described_class.new(nil)
+
+      expect(custom_response.ok?).to be(false)
+    end
+  end
+
+  describe '#body' do
+    it 'returns key error message if 401 code is received' do
+      custom_response = described_class.new(setup_exception(401))
+
+      expect(custom_response.body).to include('API key')
+    end
+
+    it 'returns an invalid city message if 404 code is received' do
+      custom_response = described_class.new(setup_exception(404))
+
+      expect(custom_response.body).to include('invalid city')
+    end
+
+    it 'returns exception message on any other response codes' do
+      custom_response = described_class.new(setup_exception(408))
+
+      expect(custom_response.body).to include('408 Request Timeout')
+    end
+  end
+
+  def setup_exception(status_code)
+    stub_request(:get, 'http//test').to_return(status: status_code)
+    RestClient.get('http//test')
+  rescue RestClient::ExceptionWithResponse => e
+    e
+  end
```

Comment:
You can do the same here, the `RestClient::ExceptionWithResponse` [has a constructor that allows you to pass the status code in the response](https://www.rubydoc.info/github/rest-client/rest-client/RestClient/Exception#initialize-instance_method):

```ruby
def failure_response(status_code)
  response = instance_double('HTTP Response', code: status_code)
  described_class(RestClient::ExceptionWithResponse.new(response))
end
```

See for example [the class where they test the RestClient::ExceptionWithResponse](https://github.com/rest-client/rest-client/blob/2c72a2e77e2e87d25ff38feba0cf048d51bd5eca/spec/unit/exceptions_spec.rb#L35).